tests/virt_autotest/login_console.pm

```diff
@@ -30,6 +30,27 @@ sub set_ssh_console_timeout_before_use {
     send_key 'ret';
 }
 
+#Just only match bootmenu-xen-kernel needle was not enough for xen host if got Xen domain0 kernel panic(bsc#1192258)
+#Need to double-check xen role after matched bootmenu-xen-kernel needle successfully
+sub double_check_xen_role {
+    record_info 'INFO', 'Double-check xen kernel';
+    if (script_run('lsmod | grep xen') == 0) {
+        diag("Boot up xen kernel successfully");
+    }
+    else {
+        record_info 'INFO', 'Check Xen hypervisor as Grub2 menuentry';
+        die 'Check Xen hypervisor as Grub2 menuentry failed' if (script_run('grub2-once --list | grep Xen') != 0);
+        save_screenshot;
+        die 'Double-check xen kernel failed';
+        save_screenshot;
```

Comment:
minor issue: save_screenshot would be redundant as the test has already died...

Reply:
sure, had removed this `save_screenshot; `